src/ProductConstructionService/ProductConstructionService.Api/Controllers/GitHubWebhookEventProcessor.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using Maestro.Data;
+using Microsoft.DotNet.GitHub.Authentication;
+using Octokit.Internal;
+using Octokit;
+using Octokit.Webhooks;
+using Octokit.Webhooks.Events;
+using Octokit.Webhooks.Events.Installation;
+using Octokit.Webhooks.Events.InstallationRepositories;
+using Microsoft.EntityFrameworkCore;
+
+namespace ProductConstructionService.Api.Controllers;
+
+public class GitHubWebhookEventProcessor : WebhookEventProcessor
+{
+    private readonly ILogger<GitHubWebhookEventProcessor> _logger;
+    private readonly IGitHubTokenProvider _tokenProvider;
+    private readonly BuildAssetRegistryContext _context;
+
+    public GitHubWebhookEventProcessor(ILogger<GitHubWebhookEventProcessor> logger, IGitHubTokenProvider tokenProvider, BuildAssetRegistryContext context)
+    {
+        _logger = logger;
+        _tokenProvider = tokenProvider;
+        _context = context;
+    }
+
+    protected override async Task ProcessInstallationWebhookAsync(WebhookHeaders headers,
+        InstallationEvent installationEvent,
+        InstallationAction action)
+    {
+        switch(installationEvent.Action)
+        {
+            case InstallationActionValue.Deleted:
+                await RemoveInstallationRepositoriesAsync(installationEvent.Installation.Id);
+                break;
+            case InstallationActionValue.Created:
+                await SynchronizeInstallationRepositoriesAsync(installationEvent.Installation.Id);
+                break;
+            default:
+                _logger.LogError("Received unknown installation action `{action}`", installationEvent.Action);
+                break;
+        }
+    }
+
+    protected override async Task ProcessInstallationRepositoriesWebhookAsync(WebhookHeaders headers, InstallationRepositoriesEvent installationRepositoriesEvent, InstallationRepositoriesAction action)
+    {
+        await SynchronizeInstallationRepositoriesAsync(installationRepositoriesEvent.Installation.Id);
+    }
+
+    private async Task RemoveInstallationRepositoriesAsync(long installationId)
+    {
+        foreach (Maestro.Data.Models.Repository repo in await _context.Repositories.Where(r => r.InstallationId == installationId).ToListAsync())
+        {
+            _context.Update(repo);
+            repo.InstallationId = 0;
```

Comment:
Should these rows be in the other order?

Reply:
Honestly, I copied the logic from the previous implementation. Now that I look at it, it does look wrong